File: AEDs2/Trabalho-Pratico-01/Palindromo/PalindromoRecursivo.c
#include <stdio.h>
#include <string.h>

int verifica(char x[500], int c, int f){
    int vale = 0;

    if(c>=f){
        return 1;
    }
    else{
        if(x[c] == x[f-1]){
            vale = verifica(x, c+1, f-1);
            return vale;
        }else{
            vale = 0;
            return vale;
        }
    }
}

int main() {
    char x[500];  
    int res = 1;
    while (res == 1) {
        scanf("%[^\n]%*c", x);
    
        if (x[0] == 'F' && x[1] ==  'I' && x[2] == 'M')
        {
            res = 0;
        }
        else {
            int vale = verifica(x, 0, strlen(x));
            if (vale == 1) {
                printf("SIM\n");
            } else {
                printf("NAO\n");
            }
        }
    }

    return 0;
}

File: AEDs2/Trabalho-Pratico-01/Palindromo/PalindromoRecursivo.java
import java.util.Scanner;

/*
    Crie um método recursivo que recebe uma string como parâmetro e retorna true se essa é um palíndromo.


    Na saída padrão, para cada linha de entrada, escreva uma linha de saída com SIM/NAO indicando se a linha é um palíndromo.


    Destaca-se que uma linha de entrada pode ter caracteres não letras


    A entrada termina com a leitura de FIM
 */

public class PalindromoRecursivo {

    public static boolean verifica(String x){
        if(x.length() == 1 || x.length() == 0){
            return true;
        }else{
            if(x.charAt(0) == x.charAt(x.length()-1)){
                return verifica(x.substring(1, x.length()-1));
            }else{
                return false;
            }
        }
    }
    
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);

        int res = 1;
        do {
            String x = sc.nextLine();
            boolean vale;

            if (x.compareTo("FIM") == 0) {
                res = 0;
            }else {
                vale = verifica(x);
                if (vale == true) {
                    System.out.println("SIM");
                } else {
                    System.out.println("NAO");
                }
            }
        } while (res == 1);
        sc.close();
    }
}
